lists/linkedlist.py

```diff
@@ -0,0 +1,175 @@
+class LinkedList:
+    def __init__(self):
+        """
+            Create a new LinkedList
+        """
+        self.head = None
+
+
+    def insert(self, element, pos):
+        """
+            Insert an element into this list
+
+            element: The element to insert
+            pos: The position to insert (0 being the start)
+
+            return: The modified LinkedList
+            raise: IndexError if pos is outside of the list
+        """
+        if pos == 0 and self.head == None:
+            self.head = element
+        elif pos == 0 and self.head != None:
+            self.head.previous = element
+            element.next = self.head
+            self.head = element
+        else:
+            counter = 0
+            testElement = self.head
+            
+            while counter < pos:
+                if testElement.next == None:
+                    raise IndexError("IndexError")
+                testElement = testElement.next
+                counter += 1
+            # self.insert(element)
+            testElement.previous.next = element
```

Comment:
You can use the insert method you wrote for `LinkedListElement` for this. You don't have to repeat yourself.